src/gui/editorwidgets/qgstexteditwrapper.cpp

```diff
@@ -74,6 +77,18 @@ QVariant QgsTextEditWrapper::value() const
     // input rather then discarding it entirely
     return QVariant( v.left( field().length() ) );
   }
+  else if ( field().type() == QVariant::Map )
+  {
+    if ( json::accept( v.toUtf8() ) )
+    {
+      QVariant qjson = QgsJsonUtils::parseJson( v.toStdString() );
+      return qjson;
+    }
+    else
+    {
+      return res;
```

Comment:
Just watched over that again. I'm not complete happy about this fallback. If the entered value is not a valid JSON value it stores it as a string. The database e.g. with the type `json` will contain a string then. Something usually Postrgres would not accept. So what would you think about something like not accepting a value change, when it's no valid JSON. Maybe blocking storing, like a widget does it, when a constraint is not fulfilled? Or at least keep the old valid JSON value and bring an info message?
Probably we even could create a JSON edit widget with `QScintilla`... 
What is your opinion @stev-0? Sorry, I know that this PR is pending already for quite some time. But it's a nice enhancement and it would be awesome when we have this solved really cute.

Reply:
Yes, happy to look at this - it would certainly be better behaviour. We've come this far .... !